infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInternalRuntime.java

```diff
@@ -187,11 +192,17 @@ protected void internalStart(Map<String, String> startOptions) throws Infrastruc
           getContext().getEnvironment().getContainers().entrySet()) {
         checkInterruption();
         String machineName = containerEntry.getKey();
-        final DockerContainerConfig config = containerEntry.getValue();
+
+        runtimeMachines.putMachine(machineName, new DockerMachine.StartingDockerMachine());
         sendStartingEvent(machineName);
+
         try {
-          startMachine(machineName, config);
+          DockerMachine machine = startMachine(machineName, containerEntry.getValue());
```

Comment:
Let's put running machine into `runtimeMachines` here instead of in `startMachine` method. I think it will make it more clear that it will be updated.